Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -1139,6 +1219,12 @@ void RenderGrid::layoutPositionedObject(RenderBox& child, bool relayoutChildren,
 
 LayoutUnit RenderGrid::gridAreaBreadthForChildIncludingAlignmentOffsets(const RenderBox& child, GridTrackSizingDirection direction) const
 {
+    if (direction == ForRows) {
+        if (isMasonryRows(style()))
+            return child.logicalHeight() + child.verticalMarginExtent();
```

Comment:
Do we need to take into account the possibility that the child is using an orthogonal writing-mode to the grid itself?